docs/design/settings/extensions/settings-extension-example.md

```diff
@@ -0,0 +1,112 @@
+Technical details of the design of the settings extension system are described elsewhere.
+
+## Background
+Suppose you are creating a Bottlerocket variant which adds some new daemon software that you've created.
+The daemon is configured via a YAML configuration file with a web address.
+When the daemon starts, it parses its configuration file for the target address and then begins periodically querying that address for updates.
+
+You'd like the following behavior to be accomplished:
+* You can configure the daemon using the Bottlerocket settings API, the value will be stored in `settings.mydaemon.query-url`.
+* Whenever the daemon is reconfigured using the Bottlerocket API, it is automatically restarted with the new configuration.
+
+In order to do this, we need to create the following resources in our Bottlerocket variant:
+* A setting extension crate called `mydaemon-settings` to add the new value `settings.mydaemon.query-url` to the API.
+* A `template` which will be rendered using the setting value and used as a configuration for the daemon.
+* A `service` file, which tells Bottlerocket how to handle the lifecycle of our daemon process as its settings change.
+
+## Creating the Settings Extension
+The easiest way to create an extension is to create a new Rust package in your project which utilizes the `bottlerocket-settings-sdk` crate.
+
+```
+├── packages
+├── sources
+│   └── mydaemon-settings
+│       ├── Cargo.toml
+│       └── src
+│           └── main.rs
+└── variants
+
+
+# sources/mydaemon/Cargo.toml
+[package]
+name = "mydaemon-settings"
+...
+[dependencies]
+bottlerocket-settings-sdk = 1.0
+...
+```
+
+Implementing the `bottlerocket_settings_sdk::Extension` trait provides a straightforward path to tell Bottlerocket how to validate and store the desired `query-url` setting.
+
+```rust
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Clone)]
+struct MyDaemonSettingV1 {
+    query-url: Url,
+}
+
+impl SettingsModel for MyDaemonSettingV1 {
+    fn get_version() -> &'static str { /* */ }
+
+    fn set(current_value: Option<Self>, new_value: Self) -> Result<Self> { /* */ }
+    
+    // etc
+}
+```
+
+You can then refer to this model when invoking the `SettingsExtension` entrypoint to the SDK:
+
+```rust
+fn main() -> Result<()> {
+    bottlerocket_settings_sdk::SettingsExtension::with_models(vec![
+        BottlerocketSetting::<MyDaemonSettingsV1>::model(),
+    ])
+    .run_extension()
+    .context("Settings extension encountered an error.")
+}
+```
+
+
+## Configuration File Template
+In your variant repository, you will need to create an RPM package which installs `mydaemon` into your custom Bottlerocket variant.
+In this package, you will also want to include a templated configuration file, which is what the settings system interacts with:
+
+```
+├── packages
+│   └── mydaemon
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── mydaemon.spec
```

Comment:
There's enough boilerplate including main.rs etc that I'm wondering if in the future sodapop or the internal build system could generate, at least the package.